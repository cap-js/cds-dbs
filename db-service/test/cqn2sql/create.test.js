'use strict'
const cds = require('@sap/cds/lib')
const cqn2sql = require('../../lib/cqn2sql')
const cqn = require('./cqn.js')

beforeAll(async () => {
  cds.model = await cds.load(__dirname + '/testModel').then(cds.linked)
  //csn = await getTestModel('testModel')
})

describe('create with select statements', () => {
  xtest('Generate SQL from CREATE stmt with entity name + as SELECT stmt', () => {
    const cqnCreate = {
      CREATE: {
        entity: 'Foo',
        as: cqn.select,
      },
    }
    const { sql } = cqn2sql(cqnCreate)
    expect({ sql }).toMatchSnapshot()
    // as SELECT is missing in sql
  })

  test('Generate SQL from CREATE stmt with entity name ', () => {
    const cqnCreate = {
      CREATE: {
        entity: 'Foo',
      },
    }
    const { sql } = cqn2sql(cqnCreate)
    expect({ sql }).toMatchSnapshot()
  })

  xtest('Generate SQL from CREATE stmt with CSN entity', () => {
    const cqnCreate = {
      CREATE: {
        entity: cds.model.definitions['Books'],
      },
    }
    const { sql } = cqn2sql(cqnCreate)
    expect({ sql }).toMatchSnapshot()
    //TypeError: Cannot read properties of undefined (reading 'length')
  })
})
