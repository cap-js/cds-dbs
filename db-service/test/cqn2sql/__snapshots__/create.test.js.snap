// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create with select statements Generate SQL from CREATE stmt with entity name  1`] = `
{
  "sql": "CREATE TABLE Foo ( ID INT, a NVARCHAR(5000), b NVARCHAR(5000), c NVARCHAR(5000), x INT )",
}
`;
